packages/instruction-plans/src/__tests__/transaction-planner-base-test.ts

```diff
@@ -0,0 +1,1652 @@
+            const planner = createBaseTransactionPlanner({ createTransactionMessage });
+
+            const instructionA = instruction('A', txPercent(25));
+            const messagePackerB = createMessagePackerInstructionPlan(txPercent(50));
+            const instructionC = instruction('C', txPercent(25));
+
+            await expect(
+                planner(
+                    parallelInstructionPlan([
+                        singleInstructionPlan(instructionA),
+                        sequentialInstructionPlan([messagePackerB, singleInstructionPlan(instructionC)]),
+                    ]),
+                ),
+            ).resolves.toEqual(
+                singleTransactionPlan([instructionA, messagePackerB.get(0, txPercent(50)), instructionC]),
+            );
+        });
+
+        /**
+         *  [Seq] ─────────────────────────▶ [Tx: A + B(1, 50%) + C]
+         *   │
+         *   ├── [A: 25%]
+         *   └── [NonDivSeq]
+         *        ├── [B(x, 50%)]
+         *        └── [C: 25%]
+         */
+        it('uses the whole non-divisible sequential message packer instruction plan when it fits in the parent sequential candidate', async () => {
+            expect.assertions(1);
+            const createTransactionMessage = createMockTransactionMessage;
+            const { instruction, singleTransactionPlan, txPercent } = getHelpers(createTransactionMessage);
+            const planner = createBaseTransactionPlanner({ createTransactionMessage });
+
+            const instructionA = instruction('A', txPercent(25));
+            const messagePackerB = createMessagePackerInstructionPlan(txPercent(50));
+            const instructionC = instruction('C', txPercent(25));
+
+            await expect(
+                planner(
+                    sequentialInstructionPlan([
+                        singleInstructionPlan(instructionA),
+                        nonDivisibleSequentialInstructionPlan([messagePackerB, singleInstructionPlan(instructionC)]),
+                    ]),
+                ),
+            ).resolves.toEqual(
+                singleTransactionPlan([instructionA, messagePackerB.get(0, txPercent(50)), instructionC]),
+            );
+        });
+    });
+
+    describe('complex scenarios', () => {
+        /**
+         *  [Par] ───────────────────────────▶ [Par]
+         *   │                                 │
+         *   ├── [Seq]                         ├── [Tx: A + B]
+         *   │    ├── [A: 40%]                 └── [NonDivSeq]
+         *   │    └── [B: 40%]                      ├── [Tx: C + D + E + G]
+         *   ├── [NonDivSeq]                        └── [Tx: F]
+         *   │    ├── [Par]
+         *   │    │    ├── [C: 25%]
+         *   │    │    └── [D: 25%]
+         *   │    └── [Par]
+         *   │         ├── [E: 25%]
+         *   │         └── [F: 50%]
+         *   └── [G: 25%]
+         */
```

Comment:
It's notable that adding `G` to the non-divisible-sequential plan increases its chance of failing to the degree that `G` can fail. Efficiency goes up, but P(success) of the `NonDivSeq` goes down.